libp2p/protocols/secure/secio.nim

```diff
@@ -175,40 +175,37 @@ proc macCheckAndDecode(sconn: SecioConn, data: var seq[byte]): bool =
   result = true
 
 proc readRawMessage(conn: Connection): Future[seq[byte]] {.async.} =
-  var lengthBuf: array[4, byte]
-  await conn.stream.readExactly(addr lengthBuf[0], lengthBuf.len)
-  let length = uint32.fromBytesBE(lengthBuf)
-  trace "Recieved message header", header = lengthBuf.shortLog, length = length
-
-  if length <= SecioMaxMessageSize: # Verify length before casting!
-    var buf = newSeq[byte](int(length))
-    if buf.len > 0: # TODO end-of-stream vs 0-byte message
+  while true: # Discard 0-length payloads
```

Comment:
Yeah I always got puzzled by the amount of callbacks/handlers we have while a `while true` is often enough  with async :) 👍